modules/ingest-user-agent/src/main/java/org/elasticsearch/ingest/useragent/UserAgentParser.java

```diff
@@ -96,8 +111,8 @@ private Pattern compilePattern(String regex, String regex_flag) {
         }
     }
 
-    private List<Map<String, String>> readParserConfigurations(XContentParser yamlParser) throws IOException {
-        List <Map<String, String>> patternList = new ArrayList<>();
+    public static List<Map<String, String>> readParserConfigurations(XContentParser yamlParser) throws IOException {
```

Comment:
Package private:
```suggestion
    static List<Map<String, String>> readParserConfigurations(XContentParser yamlParser) throws IOException {
```

Reply:
DeviceType parser is also using this method, this is why it's public.

Reply:
Since `DeviceTypeParser` is in the same package (`org.elasticsearch.ingest.useragent`), it will still be able to call that method if it is package private. We generally reserve public for methods intended for use by clients of the class outside of the package or module.